### PR TITLE
 Sort and specifying the articles

### PR DESCRIPTION
**Pull request by hira-iqbal**
-  Working of active record methods
- Use of pluck command use of  where method
- Difference between find, find_by and where clause
- Use of order method to sort the record

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -40,7 +40,10 @@ def destroy
   private
 
   def set_article
-    @article = Article.find(params[:id])
+    @article = Article.archived.find_by(id: params[:id])
+    if @article.nil?
+      redirect_to articles_path, alert: "You cannot edit a public or a private article"
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb:45]: Did you tested this use case? If yes, please add a screenshot.

**Comment by hira-iqbal** [comment]
- Working of active record methods
- Use of pluck command use of where method
- Difference between find, find_by and where clause
- Use of order method to sort the record
-

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -1,10 +1,11 @@
 class ArticlesController < ApplicationController
+ # child < parent
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb]: Remove comment please.

File: app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 before_action :set_article, only: [:show, :edit, :update, :destroy]
 http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show]
 
 
   def index
-    @articles = Article.all
+    @articles = Article.all.ordered
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb]: Try ```Article.ordered```

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (hira-iqbal) [reply app/controllers/articles_controller.rb:45]: Screenshot 2021-11-30 at 1 29 01 PM

**Review by aliammaar — APPROVED** [review]

**Pull request merged**